libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -519,6 +547,21 @@ AP_Camera::CamTrigType AP_Camera::get_trigger_type(void)
     }   
 }
 
+void AP_Camera::update_offset_location()
+{
+    offset_location = current_loc;
+    if (_cam_positions.get().is_zero()) {
+        return;  // don't wast time here
+    }
+    const Matrix3f Tbn = AP::ahrs().get_rotation_body_to_ned();
+    // Compute camera position relative to IMU
+    const Vector3f accel_body_offset = AP::ins().get_imu_pos_offset(AP::ahrs().get_primary_accel_index());
```

Comment:
Why do you need this? This is accounted for by the EKF already (in theory), so why not just have a single offset and skip the difference here? This actually looks like a way to just accumulate error by accident.

Reply:
I get that from the precland code

Reply:
Yeah, but we should just go to the center position, and allow the IMU offsets to be managed separately.